Framework/Beamline/inc/MantidBeamline/ComponentInfo.h

```diff
@@ -40,14 +44,26 @@ class MANTID_BEAMLINE_DLL ComponentInfo {
 private:
   boost::shared_ptr<std::vector<size_t>> m_assemblySortedDetectorIndices;
   boost::shared_ptr<const std::vector<std::pair<size_t, size_t>>> m_ranges;
+  Mantid::Kernel::cow_ptr<std::vector<Eigen::Vector3d>> m_positions;
+  Mantid::Kernel::cow_ptr<std::vector<Eigen::Quaterniond>> m_rotations;
   const size_t m_size = 0;
+  boost::shared_ptr<DetectorInfo> m_detectorInfo;
+  bool isDetectorDomain(const size_t componentIndex) const;
```

Comment:
Maybe not relevant immediately, but eventually this could be made public, and named `isDetector`?